modules/tinymce/src/themes/silver/main/ts/ui/general/Checkbox.ts

```diff
@@ -37,16 +36,10 @@ export const renderCheckbox = (spec: CheckboxSpec, providerBackstage: UiFactoryB
       Disabling.config({
         disabled: () => !spec.enabled || providerBackstage.isDisabled(),
         onDisabled: (component) => {
-          if (isEnabled) { // Prevent recursive loop
-            isEnabled = false;
-            Traverse.parent(component.element).each((element) => component.getSystem().getByDom(element).each(Disabling.disable));
-          }
+          Traverse.parentElement(component.element).each((element) => Class.add(element, 'tox-checkbox--disabled'));
         },
         onEnabled: (component) => {
-          if (!isEnabled) { // Prevent recursive loop
-            isEnabled = true;
-            Traverse.parent(component.element).each((element) => component.getSystem().getByDom(element).each(Disabling.enable));
-          }
+          Traverse.parentElement(component.element).each((element) => Class.remove(element, 'tox-checkbox--disabled'));
```

Comment:
I would even say there is no need for this. We can accomplish the same thing by changing the css, instead of using `tox-checkbox--disabled` we can use `:has(input:disabled)` and we won't need to run extra js to toggle css classes.

Reply:
I think `:has` selector could give us some trouble with FireFox: https://caniuse.com/?search=%3Ahas

Reply:
Seems like using the current method, and let the css stay in place, might be the best option for the time being.